### ISSUE TITLE
Exception thrown when applying the project config: Column 'groupId' cannot be null.

### ISSUE THREAD

**Issue by JorgeAnzola** [craftcms/cms#9067]
### Description
When I try to run `project-config:apply` (Or even through the admin UI)
I get the following error

```
adding sites.eb11e6c8-31ed-4454-95f5-5de3b343c4d5 ... 
error: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'groupId' cannot be null
The SQL being executed was: INSERT INTO `sites` (`uid`, `groupId`, `name`, `handle`, `language`, `hasUrls`, `baseUrl`, `primary`, `sortOrder`, `enabled`, `dateCreated`, `dateUpdated`) VALUES ('eb11e6c8-31ed-4454-95f5-5de3b343c4d5', NULL, '8-bit Llama Designs', 'default', 'en', 1, '$PRIMARY_SITE_URL', 1, 1, 1, '2021-06-21 19:34:31', '2021-06-21 19:34:31')
```

I'm not sure what to do, I rebuild the project config on production and downloaded the files, but I still get those error.
Could you shed some light? Thanks!

**Comment by brandonkelly** [comment]
Usually when this sort of ting happens, it’s because the project config and database got out of sync somehow. Try SSHing into your production server and then run

```bash
php craft project-config/rebuild
```

Then grab a fresh database backup from production and restore it to all other environments, so everything is back on the same page.

**Comment by JorgeAnzola** [comment]
Ok, so, I managed to fix the issue. I'll do my best to explain what I did just in case it helps somebody.

The error said `adding sites.eb11e6c8-31ed-4454-95f5-5de3b343c4d5`, I looked for a file containing that UID in my repo, but I couldn’t found anything. 

Something with that name showed up only on my local storage (git ignored, therefore, it was not on the server).
I went to the production database, in the table `projectconfignames`, and yes, there was a row with that UID... I deleted it. and now it works.

That helped me, but I have no idea why or what mess I had. I'll leave the possible explanation to smarter people 😄 

I'll consider the issue to be closed.
Thanks for the reply, Brandon!
